Assets/Scripts/AnimationAndMovementController.cs

```diff
@@ -4,206 +4,233 @@
-
-  // Variables to store optimized setter/getter parameter IDs
-  int isWalkingHash;
-  int isRunningHash;
-
-
-  // Variables to store player input values
-  Vector3 currentMovementInput;
-  Vector3 currentMovement;
-  Vector3 currentRunMovement;
-  bool isMovementPressed;
-  bool isRunPressed;
-
-  // Constants
-  float rotationFactorPerFrame = 15.0f;
-  float walkMultiplier = 3.0f;
-  float runMultiplier = 7.0f;
-  int zero = 0;
-  float gravity = -9.8f;
-  float groundedGravity = -.05f;
-
-  // Jumping variables
-  bool isJumpPressed = false;
-  float initialJumpVelocity;
-  float maxJumpHeight = 6.0f;
-  float maxJumpTime = 0.75f * 1.2f;
-  bool isJumping = false;
-  int isJumpingHash;
-  bool isJumpAnimating = false;
-
-  // Awake is called earlier than start
-  void Awake()
-  {
-    // Set initial reference variables
-    playerInput = new PlayerInput();
-    characterController = GetComponent<CharacterController>();
-    animator = GetComponent<Animator>();
-
-    // Set parameter hash references
-    isWalkingHash = Animator.StringToHash("isWalking");
-    isRunningHash = Animator.StringToHash("isRunning");
-    isJumpingHash = Animator.StringToHash("isJumping");
-
-    // Set player input callbacks
-    playerInput.CharacterControls.Move.started += onMovementInput;
-    playerInput.CharacterControls.Move.canceled += onMovementInput;
-    playerInput.CharacterControls.Move.performed += onMovementInput;
-    playerInput.CharacterControls.Run.started += onRun;
-    playerInput.CharacterControls.Run.canceled += onRun;
-    playerInput.CharacterControls.Jump.started += onJump;
-    playerInput.CharacterControls.Jump.canceled += onJump;
-
-    setupJumpVariables();
-  }
-
-  void setupJumpVariables()
-  {
-    float timeToApex = maxJumpTime / 2;
-    gravity = (-2 * maxJumpHeight) / Mathf.Pow(timeToApex, 2);
-    initialJumpVelocity = (2 * maxJumpHeight) / timeToApex;
-  }
-
-  void onJump(InputAction.CallbackContext context)
-  {
-    isJumpPressed = context.ReadValueAsButton();
-    Debug.Log(isJumpPressed);
-  }
-
-  void onRun(InputAction.CallbackContext context)
-  {
-    isRunPressed = context.ReadValueAsButton();
-  }
-
-  void onMovementInput(InputAction.CallbackContext context)
-  {
-    currentMovementInput = context.ReadValue<Vector2>();
-    currentMovement.x = currentMovementInput.x * walkMultiplier;
-    currentMovement.z = currentMovementInput.y * walkMultiplier;
-    currentRunMovement.x = currentMovementInput.x * runMultiplier;
-    currentRunMovement.z = currentMovementInput.y * runMultiplier;
-    isMovementPressed = currentMovementInput.x != zero || currentMovementInput.y != zero;
-    Debug.Log(currentMovementInput);
-  }
-
-  void handleAnimation()
-  {
-    bool isWalking = animator.GetBool(isWalkingHash);
-    bool isRunning = animator.GetBool(isRunningHash);
+    void Update()
+    {
+        handleAnimation();
+        handleRotation();
+        handleGravity();
+        handleJump();
+        move();
```

Comment:
this isn't as drastic as it looks! i mainly just pulled the logic out to make these composed methods for readability.